packages/odata-service-inquirer/src/prompts/connectionValidator.ts

```diff
@@ -590,16 +590,23 @@ export class ConnectionValidator {
         this.resetValidity();
         // Get the destination URL in the BAS specific form <protocol>://<destinationName>.dest
         const destUrl = getDestinationUrlForAppStudio(destination.Name, servicePath);
-        // Get the destination URL in the portable form <protocol>://<host>:<port>
-        this._destinationUrl = servicePath ? new URL(`${destination.Host}${servicePath}`).toString() : destination.Host;
+        // Get the destination URL in the portable form <protocol>://<host>:<port>.
+        // We remove trailing slashes (up to 10, infinite would allow DOS attack) from the host to avoid double slashes when appending the service path.
+        this._destinationUrl = servicePath
+            ? destUrl.replace(`https://${destination.Name}.dest`, destination.Host.replace(/\/{1,10}$/, ''))
+            : destination.Host;
         this._destination = destination;
         // No need to apply sap-client as this happens automatically (from destination config) when going through the BAS proxy
         const status = await this.checkUrl(new URL(destUrl), undefined, undefined, {
             odataVersion: requiredOdataVersion
         });
-        this._validatedUrl = destUrl;
+
         const validationResult = this.getValidationResultFromStatusCode(status);
 
+        if (this.validity.reachable && (!this.validity.authRequired || this.validity.authenticated)) {
```

Comment:
Does this handle the scearnio where for a SAMLAssertion destination, a HTTP 401 shouldn't prompt the user for their credentials?

Reply:
Not sure I follow the question. There is no prompting here. That would be a different part of the code . This is to ensure we dont add the url as validated if it is not reachable or needs auth.

Reply:
Thanks @IainSAP I will create a separate issue.